src/solvers/refinement/string_refinement.cpp

```diff
@@ -2234,6 +2234,41 @@ exprt string_refinementt::get(const exprt &expr) const
   exprt ecopy(expr);
   (void)symbol_resolve.replace_expr(ecopy);
 
+  // Special treatment for index expressions
+  const auto &index_expr = expr_try_dynamic_cast<index_exprt>(ecopy);
+  if(index_expr && is_char_type(index_expr->type()))
+  {
+    std::reference_wrapper<const exprt> current(index_expr->array());
+    while(current.get().id() == ID_if)
+    {
+      const auto &if_expr = expr_dynamic_cast<if_exprt>(current.get());
+      const exprt cond = get(if_expr.cond());
+      if(cond.is_true())
```

Comment:
`if(get(if_expr.cond()).is_true())` - I like oneliners.

Reply:
we also use `cond` with `.is_false` after

Reply:
👌